src/build_system.ml

```diff
@@ -443,8 +444,18 @@ let compile_rule t ~all_targets_by_dir ?(allow_override=false) pre_rule =
           acc || prev_hash <> hash)
     in
     let targets_min_ts = min_timestamp t targets_as_list  in
-    let deps_max_ts    = max_timestamp t all_deps_as_list in
-    if rule_changed ||
+    let deps_max_ts = max_timestamp t all_deps_as_list in
+    let is_runtest =
+      Pset.for_all targets ~f:(fun target ->
+        match Path.extract_build_context target with
+        | Some (".aliases", target) ->
+            (target
+            |> Path.basename
+            |> String.is_prefix ~prefix:"runtest-")
```

Comment:
This would also catch aliases `runtest-XXX` which a user might define. The proper way is to chop the suffix. I added a fonction `Alias.name_of_file` to deal with this